source/extensions/upstreams/http/http/upstream_request.h

```diff
@@ -82,6 +82,12 @@ class HttpUpstream : public Router::GenericUpstream, public Envoy::Http::StreamC
     request_encoder_->getStream().resetStream(Envoy::Http::StreamResetReason::LocalReset);
   }
 
+  void setAccount(Buffer::AccountSharedPtr account) override {
+    // TODO(kbaichoo): check std::move() shared_ptr avoids creating new one esp.
+    // if just copying... like to avoid to atomic inc, other bookkeeping etc. if possible.
```

Comment:
std::move does avoid the atomic inc/dec on shared_ptr

The other option to consider is const shared_ptr&